src/main/java/reporter/AutoReporter.java

```diff
@@ -0,0 +1,7 @@
+package reporter;
+
+public interface AutoReporter<J> {
+    String makeReportFrom(J something);
+
+    void makeReportFromTo(J something1, J something2);
```

Comment:
```suggestion
public interface ReportService<J> {
    void makeReport(J source, J target);
```

- remember about correct and informative naming: why `Auto`Reporter? 
- do not use digits in your variable names
- do not write methods that can be not used - you should follow the Interface Segregation principle